pkg/apis/v1/nodepool_validation_cel_test.go

```diff
@@ -697,4 +700,148 @@ var _ = Describe("CEL/Validation", func() {
 			Expect(env.Client.Create(ctx, nodePool)).ToNot(Succeed())
 		})
 	})
+	Context("Replicas", func() {
+		Context("Valid Replicas Values", func() {
+			It("should succeed when replicas is set to a positive value", func() {
+				nodePool.Spec.Replicas = lo.ToPtr(int64(5))
+				Expect(env.Client.Create(ctx, nodePool)).To(Succeed())
+				Expect(nodePool.RuntimeValidate(ctx)).To(Succeed())
+			})
+
+			It("should succeed when replicas is set to zero", func() {
+				nodePool.Spec.Replicas = lo.ToPtr(int64(0))
+				Expect(env.Client.Create(ctx, nodePool)).To(Succeed())
+				Expect(nodePool.RuntimeValidate(ctx)).To(Succeed())
+			})
+
+			It("should succeed when replicas is set to a large value", func() {
+				nodePool.Spec.Replicas = lo.ToPtr(int64(1000))
+				Expect(env.Client.Create(ctx, nodePool)).To(Succeed())
+				Expect(nodePool.RuntimeValidate(ctx)).To(Succeed())
+			})
+		})
+
+		Context("Invalid Replicas Values", func() {
+			It("should fail when replicas is set to a negative value", func() {
+				nodePool.Spec.Replicas = lo.ToPtr(int64(-100))
+				nodePool.Spec.Disruption.ConsolidateAfter = MustParseNillableDuration("0s")
+				Expect(env.Client.Create(ctx, nodePool)).ToNot(Succeed())
+			})
+		})
+
+		Context("Compatible Fields", func() {
+			It("should succeed with replicas and comprehensive static nodepool configuration", func() {
```

Comment:
I would use a `DescribeTable` and have an individual entry for each other field you'd like to verify with. This way, if there is a regression the entry that fails indicates the exact issue.